### DISCUSSION TITLE
How does Infinity score so well in the benchmarks compared to other vector indexes like Qdrant

### DISCUSSION THREAD

**Discussion by gokturkDev** [infiniflow/infinity#1479] [General]
I can't wrap my head around on how much faster the infinity is compared to other vector indexes in the benchmark. The docs don't mention any reason why infinity scores better in terms of QPS or latency. Are there any tradeoffs at play here or is it just due to better implementation/algorithms?

**Comment by JinHai-CN** [comment]
1. No trade off comparing to other vector database or search engine.
2. Infinty does have better index algorithm and implementation. We may write an article to describe it in future.

**Comment by writinwaters** [comment]
Hi @gokturkDev, you may want to read this blog: https://infiniflow.org/blog/fastest-hybrid-search.

**Reply by gokturkDev** [reply]
Hi this is exactly what I needed thanks! I guess the real differentiation comes from the way HNSW is built and the extensive quantization. I think this also explains why infinity's recall is 0.97 and QPS is 16K while Elastic has 0.992 recall and only 900 QPS in SIFT1M

I guess it boils down to whether you can afford a minor decrease in accuracy to get massive speed boosts. Wdyt @writinwaters @JinHai-CN ?

**Reply by JinHai-CN** [reply]
Hi @gokturkDev There might be some misunderstanding? The SIFT1M benchmark is actually made when Infinity and ElasticSearch have the **same** recall, which is 0.97. We also tested Infinity which has 0.992 recall.

| Recall     | 0.972     | 0.992     |
| -------- | -------- | -------- |
| Infinity (QPS)| 16K | 8K |
| Elasticsearch (QPS)| 0.9K |Not tested |

**Reply by gokturkDev** [reply]
Wow great article and great work @JinHai-CN !
